CONTRIBUTING.md

```diff
@@ -0,0 +1,203 @@
+=================================
+
+Developing Open Source is great fun!  Join us on the [image.sc forums](https://forum.image.sc) and tell us
+which of the following challenges you'd like to solve.
+
+* Mentoring is available for those new to scientific programming in Python.
+* If you're looking for something to implement or to fix, you can browse the
+  [open issues on GitHub](https://github.com/TheJacksonLaboratory/ezomero/issues?q=is%3Aopen).
+* The technical detail of the development process is summed up below.
+
+
+Stylistic Guidelines
+--------------------
+
+* Functions should strive to, whenever possible, have arguments and returns in common Python types, or in numpy/scipy types. No passing OMERO objects back and forth!
+
+* If there is a function from the base [OMERO-py API](https://downloads.openmicroscopy.org/omero/5.6.3/api/python/) that does something, we do not need to duplicate it. Wrapping OMERO-py functions for simpler input/output is fine.
+
+* Think twice before adding dependencies. Do you need to? Can you do the same thing with base types and/or numpy?
+
+* Keep things simple. ezomero is not supposed to be a fully-featured OMERO API; it is explicitly designed to be an easy-to-use API. If your function is not easy to use, it probably does not fit here.
+
+* Set up your editor to remove trailing whitespace.  Follow [PEP08](https://www.python.org/dev/peps/pep-0008/).  Check code with pyflakes / flake8.
+
+* Use relative module imports, i.e. ``from ._misc import xyz`` rather than
+  ``from ezomero._misc import xyz``.
+  
+  
+Development process
+-------------------
+
+Here's the long and short of it:
+
+1. If you are a first-time contributor:
+
+   * Go to [https://github.com/TheJacksonLaboratory/ezomero](https://github.com/TheJacksonLaboratory/ezomero) and click the
+     "fork" button to create your own copy of the project.
+
+   * Clone the project to your local computer:
+
+      `git clone https://github.com/your-username/ezomero.git`
+
+   * Change the directory:
+
+      `cd ezomero`
+
+   * Add the upstream repository:
+
+      `git remote add upstream https://github.com/TheJacksonLaboratory/ezomero.git`
+
+   * Now, you have remote repositories named:
+
+     - ``upstream``, which refers to the ``ezomero`` repository
+     - ``origin``, which refers to your personal fork
+
+
+2. Develop your contribution:
+
+   * Pull the latest changes from upstream:
+
+      `git checkout main`
+      `git pull upstream main`
+
+   * Create a branch for the feature you want to work on. Since the
+     branch name will appear in the merge message, use a sensible name
+     such as 'add-retrieve-tags':
+
+      `git checkout -b add-retrieve-tags`
+
+   * Commit locally as you progress (``git add`` and ``git commit``)
+
+3. To submit your contribution:
+
+   * Push your changes back to your fork on GitHub::
+
+      git push origin add-retrieve-tags
+
+   * Enter your GitHub username and password (repeat contributors or advanced
```

Comment:
note that GitHub has deprecated password authenticating via git without an access token and will discontinue this functionality by Aug 13, 2021. So we might mention this

Reply:
This change (and the two previous comments) implemented in a new commit